packages/steiger-plugin-fsd/src/no-segments-on-sliced-layers/index.spec.ts

```diff
@@ -0,0 +1,99 @@
+import { describe, expect, it } from 'vitest'
+
+import noSegmentsOnSlicedLayers from './index.js'
+import { joinFromRoot, parseIntoFsdRoot } from '../_lib/prepare-test.js'
+
+describe('no-segments-on-sliced-layers rule', () => {
+  it('reports no errors on a project where the sliced layers has no segments in direct children', () => {
+    const root = parseIntoFsdRoot(`
+      📂 shared
+        📂 ui
+          📄 index.ts
+        📂 i18n
+          📄 index.ts
+      📂 entities
+        📂 user
+          📂 ui
+            📄 Name.tsx
+          📂 api
+            📄 useCurrentUser.ts
+          📄 index.ts
+        📂 document
+          📂 api
+            📄 useDocument.ts
+      📂 pages
+        📂 home
+          📂 ui
+          📄 index.ts
+    `)
+
+    expect(noSegmentsOnSlicedLayers.check(root)).toEqual({ diagnostics: [] })
+  })
+
+  it('reports errors on a project where a sliced layer has segments among its direct children', () => {
+    const root = parseIntoFsdRoot(`
+      📂 shared
+        📂 ui
+          📄 index.ts
+        📂 i18n
+          📄 index.ts
+      📂 entities
+        📂 user
+          📄 index.ts
+          📄 Name.tsx
+        📂 ui
+          📄 index.ts
+      📂 features
+        📂 user
+          📂 ui
+            📄 LogIn.tsx
+            📄 index.ts
+          📄 index.ts
+        📂 api
+          📄 index.ts
+      📂 widgets
+        📂 footer
+          📂 ui
+            📄 Footer.tsx
+            📄 index.ts
+          📄 index.ts
+        📂 config
+          📄 index.ts
+      📂 pages
+        📂 home
+          📂 ui
+          📄 index.ts
+        📂 settings
+          📂 profile
+            📄 ProfilePage.tsx
+            📄 index.ts
+        📂 lib
+          📄 index.ts
+    `)
+
+    const diagnostics = noSegmentsOnSlicedLayers.check(root).diagnostics
```

Comment:
suggestion (non-blocking): it would be better for the test to sort these diagnostics so that sort order doesn't break the test under a different implementation. We have a helper called `compareMessages` that you can pass to `.sort()` as a comparator function

Reply:
Yes, thank you for the suggestion